presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -260,7 +264,7 @@ public BaseJdbcClient(
     @Override
     public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
     {
-        return jdbcTypeToPrestoType(session, typeHandle);
+        return jdbcTypeToPrestoType(session, typeHandle, jdbcTypesMappedToVarchar);
```

Comment:
This should be handled before calling `jdbcTypeToPrestoType`, so let's keep it inside JdbcClient and don't change that method.

Since the mapping is not that trivial (eg disabling predicate pushdown), I think a helper method could be useful. 
It could be used like that:

```
Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
if (mapping.isPresent()) {
    return mapping;
}
```

where `getForcedMappingToVarchar` is defined in BaseJdbcClient:
```
protected Optional<...> getForcedMappingToVarchar(...) { ... }
```